app/views/csvfiles/new.html.erb

```diff
@@ -0,0 +1,155 @@
+                            filestring: csvStringForDownload },
+                    success: function(data){
+                        let divAlert = document.createElement('div');
+                        divAlert.classList.add('alert');
+                        divAlert.classList.add('alert-success');
+                        divAlert.innerHTML = 'File save successfully!';
+                        headerContainer.appendChild(divAlert);
+                    },
+                    error: function(XMLHttpRequest, textStatus, errorThrown){
+                        console.log(errorThrown);
+                        let divAlert = document.createElement('div');
+                        divAlert.classList.add('alert');
+                        divAlert.classList.add('alert-danger');
+                        divAlert.innerHTML = 'There was some error in saving the file.';
+                        headerContainer.appendChild(divAlert);
+                    }
+                });
+            }
+        });
+
+        $('#' + prevFile).click(function(){
+            $.ajax({
+                url: '/graph/prev_file',
+                type: 'get',
+                data: { uid: <%= current_user.id %> },
+                success: function(data){
+                    displayPreviousFiles(data);
+                },
+                error: function(XMLHttpRequest, textStatus, errorThrown){
+                        console.log(errorThrown);
+                        let divAlert = document.createElement('div');
+                        divAlert.classList.add('alert');
+                        divAlert.classList.add('alert-danger');
+                        divAlert.innerHTML = 'There was some error in retrieving the file.';
+                        headerContainer.appendChild(divAlert);
+                }
+            });
+        });
+
+        function displayPreviousFiles(data){
+            var table=document.createElement('table');
+            for (let i = 0;i<data.length;i++){
+                let tr = document.createElement('tr');
+                let td = document.createElement('td');
+                let radio = document.createElement('input');
+                radio.type = 'radio';
+                radio.value = i;
+                radio.name = data[0]['csvfile']['uid'] + 'user_id';
+                td.appendChild(radio);
+                td.appendChild(document.createTextNode(data[i]['csvfile']['filetitle']));
+                tr.appendChild(td);
+                table.appendChild(tr);
+            }
+            let div = document.getElementById(SimpleDataGrapherObject.view.upload_button_container);
+            div.appendChild(table);
+            selectFile(data);
+        }
+        function selectFile(data){
+            $('#' + SimpleDataGrapherObject.view.uploadButtonId).click(function(){
+                let name = data[0]['csvfile']['uid'] + 'user_id';
+                let index = $('input[name='+ name +']:checked').val();
```

Comment:
whitespace around `+`

Reply:
Which one?

Reply:
the one after name= and the one before the closing square bracket

Reply:
Okay!